src/backend/distributed/metadata/node_metadata.c

```diff
@@ -753,13 +754,662 @@ PgDistTableMetadataSyncCommandList(void)
 }
 
 
+/*
+ * SendTableMetadataDeletionCommands sends metadata deletion commands
+ * within metadata connection.
+ */
+void
+SendTableMetadataDeletionCommands(List *nodeList)
+{
+	/* find relevant worker nodes */
+	List *nodesWithMetadata = NIL;
+	WorkerNode *workerNode = NULL;
+	foreach_ptr(workerNode, nodeList)
+	{
+		if (NodeIsPrimary(workerNode) && !NodeIsCoordinator(workerNode))
+		{
+			nodesWithMetadata = lappend(nodesWithMetadata, workerNode);
+		}
+	}
+
+	if (nodesWithMetadata == NIL)
+	{
+		return;
+	}
+
+	List *metadataDeletionCommands = NIL;
+	metadataDeletionCommands = lappend(metadataDeletionCommands,
+									   DELETE_ALL_PARTITIONS);
+	metadataDeletionCommands = lappend(metadataDeletionCommands,
+									   DELETE_ALL_SHARDS);
+	metadataDeletionCommands = lappend(metadataDeletionCommands,
+									   DELETE_ALL_PLACEMENTS);
+	metadataDeletionCommands = lappend(metadataDeletionCommands,
+									   DELETE_ALL_DISTRIBUTED_OBJECTS);
+	metadataDeletionCommands = lappend(metadataDeletionCommands,
+									   DELETE_ALL_COLOCATION);
+	metadataDeletionCommands = lcons(DISABLE_DDL_PROPAGATION,
+									 metadataDeletionCommands);
+	metadataDeletionCommands = lappend(metadataDeletionCommands,
+									   ENABLE_DDL_PROPAGATION);
+	SendMetadataCommandListToWorkerListInCoordinatedTransaction(nodesWithMetadata,
+																CurrentUserName(),
+																metadataDeletionCommands);
+	list_free(nodesWithMetadata);
+}
+
+
+/*
+ * SendIterableTableMetadataCommands sends all tables' metadata within metadata
+ * connection.
+ */
+void
+SendIterableTableMetadataCommands(List *nodeList)
```

Comment:
what does `Iterable` mean here?

Reply:
It is removed in later commit by `SendReplicationCommandsOutsideTransactionPerDistObject`.